pkg/daemon/daemon.go

```diff
@@ -1601,9 +1612,99 @@ func PersistNetworkInterfaces(osRoot string) error {
 		return fmt.Errorf("failed to run nmstatectl: %w", err)
 	}
 
+	// If the node is using networking in the initrd (e.g. for Tang pinning), then
+	// it's possible that the network configuration there also relies on specific
+	// network names. To be safe, we'll add kargs to have the names also apply in
+	// the initrd, and then possibly remove them afterwards as part of cleanup.
+
+	if usesInitrdNetworking, err := HostUsesNetworkingInInitrd(); err != nil {
+		return err
+	} else if !usesInitrdNetworking {
+		glog.Info("Not modifying ifname= kargs; host doesn't use initrd networking")
+		return nil // nothing more to do!
+	}
```

Comment:
I would break this to it's own separate `if ` statement rather than make it part of an `else if`. It appears a bit awkward to use a variable that was defined in the previous if condition as part of and else condition, but maybe that's because I don't read/write go every day?

Reply:
Yeah, this is pretty common in golang. The `usesInitrdNetworking` variable is scoped to the if-statement.